.github/workflows/generate-preview.yml

```diff
@@ -0,0 +1,93 @@
+name: Generate asciinema preview gif
+
+on:
+  push:
+    tags:
+      - 'v*'
+
+jobs:
+  generate:
+    runs-on: ubuntu-latest
+    env:
+      DEMO_MAGIC_INSTALL: /tmp/demo-magic
+      TERM: xterm
+    steps:
+    # Checks-out the project under $GITHUB_WORKSPACE/git-gud
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+        path: git-gud
+
+    - uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+
+    - uses: actions/setup-node@v1
+      with:
+        node-version: '14'
+
+    - name: Set up packager-managed dependencies for gif build
+      # demo-magic.sh requires [pv]
+      run: |
+        sudo apt-get update
+        python -m pip install --upgrade pip
+        pip install setuptools wheel twine
```

Comment:
Are these necessary?

Reply:
Yes, because `pip install` breaks without them, AFAIK. It's been a while since I touched that portion of code.